server/client.go

```diff
@@ -4555,32 +4557,21 @@ func (c *client) processPingTimer() {
 
 	c.Debugf("%s Ping Timer", c.kindString())
 
-	var sendPing bool
-
 	pingInterval := c.srv.getOpts().PingInterval
 	if c.kind == GATEWAY {
 		pingInterval = adjustPingIntervalForGateway(pingInterval)
 	}
 	now := time.Now()
 	needRTT := c.rtt == 0 || now.Sub(c.rttStart) > DEFAULT_RTT_MEASUREMENT_INTERVAL
 
-	// Do not delay PINGs for GATEWAY or spoke LEAF connections.
-	if c.kind == GATEWAY || c.isSpokeLeafNode() {
```

Comment:
Doing the write is most important over potential long haul or multi-hope networks. So even if we have received something we may need to write, so prefer this logic remains.